vehicle/src/Vehicle/Backend/LossFunction/Compile.hs

```diff
@@ -48,87 +52,74 @@ newtype Domain = Domain ()
+  | Min LExpr LExpr                           -- Min
+  | Max LExpr LExpr                           -- Max
+  | Addition LExpr LExpr                      -- addition
+  | Subtraction LExpr LExpr                   -- subtraction
+  | Multiplication LExpr LExpr                -- multiplication
+  | Division LExpr LExpr                      -- division
+  | IndicatorFunction LExpr LExpr             -- an indicator function
+  | Variable V.DBIndex                        -- variable (bound)
+  | FreeVariable Name                         -- variable (free)
+  | NetworkApplication Name (NonEmpty LExpr)  -- neural network
+  | Quantifier Quantifier Name Domain LExpr   -- quantifiers forall, exists
+  | At LExpr LExpr                            -- at
+  | TensorLiteral [LExpr]                     -- tensor
+  | Lambda Name LExpr                         -- lambda expression
+  | Let Name LExpr LExpr                      -- let expression
+  | Power LExpr LExpr                         -- exponential
   deriving (Eq, Ord, Generic, Show)
 
 instance FromJSON LExpr
 instance ToJSON LExpr
 
 --------------------------------------------------------------------------------
 -- Compilation
--- the translation into the LExpr
-
-compile :: MonadCompile m => V.CheckedProg -> V.PropertyContext -> NetworkContext -> m [LDecl]
-compile prog propertyCtx networkCtx = do
-  normalisedProg <- normalise prog normalisationOptions
-  runReaderT (compileProg normalisedProg) (propertyCtx, networkCtx)
-
-normalisationOptions :: NormalisationOptions
-normalisationOptions = Options
-  { declContext                 = mempty
-  , boundContext                = mempty
-  , normaliseDeclApplications   = False
-  , normaliseLambdaApplications = False
-  , normaliseStdLibApplications = False
-  , normaliseBuiltin            = normBuiltin
-  , normaliseWeakly             = False
-  }
-
-normBuiltin :: V.Builtin -> Bool
-normBuiltin b = case b of
-  V.TypeClassOp t -> case t of
-    V.FromNatTC {} -> True
-    V.FromRatTC    -> True
-    V.FromVecTC {} -> True
-    V.NotTC        -> True
-    V.AndTC        -> True
-    V.OrTC         -> True
-    V.ImpliesTC    -> True
-    V.MapTC        -> True
-    V.NegTC        -> True
-    V.AddTC        -> True
-    V.SubTC        -> True
-    V.MulTC        -> True
-    V.DivTC        -> True
+-- the translation into the LExpr (this is the exported top compile function)
 
-    _              -> False
-
-  V.FromNat {}       -> True
-  V.FromRat {}       -> True
-  V.FromVec {}       -> True
-  V.Foreach{}        -> True
+--------------------------------------------------------------------------------
+-- different avilable differentiable logics (types of translation from the constraint to loss function) are
+-- DL2
+-- Godel
+-- Lukasiewicz
+-- Product based
+-- Yager
 
-  _                  -> False
+--they can be found in Vehicle.Backend.Prelude and the default option if none is provided is DL2.
```

Comment:
Again, these comments should go next to the translations rather than here right?